File: src/services/currencyConverter.js
/**
 * Currency Conversion Utility
 * Converts between INR (Indian Rupees) and POL (Polygon)
 */

// Exchange rates (update these periodically)
// 1 POL ≈ $0.80 USD
// 1 USD ≈ 83 INR
// Therefore: 1 POL ≈ 66.4 INR (approximately)

const EXCHANGE_RATES = {
    POL_TO_USD: 0.80,  // 1 POL = $0.80 USD
    USD_TO_INR: 83,    // 1 USD = 83 INR
};

/**
 * Convert INR (Rupees) to POL
 * @param {number} amountINR - Amount in Indian Rupees
 * @returns {number} Amount in POL
 */
export const convertINRtoPOL = (amountINR) => {
    if (!amountINR || amountINR <= 0) return 0;
    
    const amountUSD = amountINR / EXCHANGE_RATES.USD_TO_INR;
    const amountPOL = amountUSD / EXCHANGE_RATES.POL_TO_USD;
    
    // Return with 4 decimal places precision
    return parseFloat(amountPOL.toFixed(4));
};

/**
 * Convert POL to INR (Rupees)
 * @param {number} amountPOL - Amount in POL
 * @returns {number} Amount in Indian Rupees
 */
export const convertPOLtoINR = (amountPOL) => {
    if (!amountPOL || amountPOL <= 0) return 0;
    
    const amountUSD = amountPOL * EXCHANGE_RATES.POL_TO_USD;
    const amountINR = amountUSD * EXCHANGE_RATES.USD_TO_INR;
    
    // Return as integer (no decimal for rupees)
    return Math.round(amountINR);
};

/**
 * Get exchange rate information
 * @returns {object} Current exchange rates and conversion info
 */
export const getExchangeRateInfo = () => {
    const polToInr = EXCHANGE_RATES.POL_TO_USD * EXCHANGE_RATES.USD_TO_INR;
    
    return {
        POL_TO_USD: EXCHANGE_RATES.POL_TO_USD,
        USD_TO_INR: EXCHANGE_RATES.USD_TO_INR,
        POL_TO_INR: polToInr.toFixed(2),
        lastUpdated: new Date().toISOString()
    };
};

/**
 * Format currency for display
 * @param {number} amount - Amount to format
 * @param {string} currency - Currency code (POL, INR, USD)
 * @returns {string} Formatted currency string
 */
export const formatCurrency = (amount, currency = "POL") => {
    switch(currency) {
        case "INR":
            return `₹${amount.toLocaleString("en-IN")}`;
        case "USD":
            return `$${amount.toFixed(2)}`;
        case "POL":
        default:
            return `${amount.toFixed(4)} POL`;
    }
};

File: backend/currency_converter.py
"""Currency conversion utilities for INR to POL conversion."""

# Exchange rates (same as frontend for consistency)
# 1 POL ≈ $0.80 USD
# 1 USD ≈ 83 INR
# Therefore: 1 POL ≈ 66.4 INR

POL_TO_USD = 0.80  # 1 POL = $0.80 USD
USD_TO_INR = 83    # 1 USD = 83 INR
POL_TO_INR = POL_TO_USD * USD_TO_INR  # 1 POL ≈ 66.4 INR


def convert_inr_to_pol(amount_inr: float) -> float:
    """
    Convert INR (Indian Rupees) to POL (Polygon)
    
    Args:
        amount_inr: Amount in Indian Rupees
    
    Returns:
        Amount in POL with 4 decimal places precision
    """
    if not amount_inr or amount_inr <= 0:
        return 0.0
    
    amount_usd = amount_inr / USD_TO_INR
    amount_pol = amount_usd / POL_TO_USD
    
    # Return with 4 decimal places precision
    return round(amount_pol, 4)


def convert_pol_to_inr(amount_pol: float) -> int:
    """
    Convert POL (Polygon) to INR (Indian Rupees)
    
    Args:
        amount_pol: Amount in POL
    
    Returns:
        Amount in INR as integer
    """
    if not amount_pol or amount_pol <= 0:
        return 0
    
    amount_usd = amount_pol * POL_TO_USD
    amount_inr = amount_usd * USD_TO_INR
    
    # Return as integer (no decimal for rupees)
    return round(amount_inr)


def get_exchange_rate_info() -> dict:
    """Get current exchange rate information"""
    return {
        "POL_TO_USD": POL_TO_USD,
        "USD_TO_INR": USD_TO_INR,
        "POL_TO_INR": round(POL_TO_INR, 2),
    }
